### ISSUE TITLE
Problems with IA models being used for long periods of time

### ISSUE THREAD

**Issue by puxd** [harfang3d/dogfight-sandbox-hg2#84]
Hi, when I train intelligences for air combat missions using reinforcement learning algorithms, the "enemy aircraft" that the intelligences pair with are outputting maneuvers using your IA model. At the beginning of the training the "enemy aircraft" is able to output normal maneuvers, but as the training time progresses it seems to have a bug and leaves the IA model! The output from the command line looks like the picture below. what can I do to solve this problem?
![question](https://github.com/harfang3d/dogfight-sandbox-hg2/assets/53037406/276d1307-914b-4611-b82a-40923a2dea4f)

**Comment by muratozbek23** [comment]
Hello, do you start the plane as IA at the beginning of each episode?

**Comment by puxd** [comment]
> Hello, do you start the plane as IA at the beginning of each episode?

No, I don't turn IA mode on every episode, I turn it on at the beginning of training. After a number of turns of training the airplane leaves IA mode on its own and goes out of control.

**Comment by muratozbek23** [comment]
I don't think there will be a problem if you do this at the beginning of each episode.

**Comment by puxd** [comment]
> I don't think there will be a problem if you turn on at the beginning of each episode.
Yes, I turned on IA and trained before the start of each turn of the mission like you wanted. But it is still not tracking after some rounds of training, as shown in the image below, where red represents the algorithm-controlled intelligences and gray represents the IA-controlled aircraft.（The second image represents the trajectory of IA when it goes wrong.）

![1](https://github.com/harfang3d/dogfight-sandbox-hg2/assets/53037406/449c8c5e-41bf-407a-b0f1-2d617cb6ead0)

![2](https://github.com/harfang3d/dogfight-sandbox-hg2/assets/53037406/5eb64f22-2bf7-4755-a0a5-0baba887d6a1)

**Comment by cym-cym** [comment]
> > I don't think there will be a problem if you turn on at the beginning of each episode.
> > Yes, I turned on IA and trained before the start of each turn of the mission like you wanted. But it is still not tracking after some rounds of training, as shown in the image below, where red represents the algorithm-controlled intelligences and gray represents the IA-controlled aircraft.（The second image represents the trajectory of IA when it goes wrong.）
> 
> ![1](https://private-user-images.githubusercontent.com/53037406/309471198-449c8c5e-41bf-407a-b0f1-2d617cb6ead0.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA0MjQxNDAsIm5iZiI6MTcxMDQyMzg0MCwicGF0aCI6Ii81MzAzNzQwNi8zMDk0NzExOTgtNDQ5YzhjNWUtNDFiZi00MDdhLWIwZjEtMmQ2MTdjYjZlYWQwLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE0VDEzNDQwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThlYThmNzk1ZjA0Yzk2ZThkNzQ3MWRkZWMwMzc4Njc3MDk0MjVhZDBjYjUyZmEzMTU1YTEzZTNkNTgyMjIyNWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Z2h6D2FrO6GqaX1AHQ27_7ki6xitg0KLTY6RhiT2wjc)
> 
> ![2](https://private-user-images.githubusercontent.com/53037406/309471203-5eb64f22-2bf7-4755-a0a5-0baba887d6a1.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA0MjQxNDAsIm5iZiI6MTcxMDQyMzg0MCwicGF0aCI6Ii81MzAzNzQwNi8zMDk0NzEyMDMtNWViNjRmMjItMmJmNy00NzU1LWEwYTUtMGJhYmE4ODdkNmExLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE0VDEzNDQwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI4NzU4YWNjYmQyZDdjZmNmYjM1ZjdhZWNkMjFkNDdlOTZiYjRlNmFmMzMyNTQ1NDA1MGViZGM1OTJhY2Q1NGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-HXglb7XPMA_zrfDiADuu2WEc454Qa-k9J0NDUtkV24)

Is your problem solved? I am also doing reinforcement learning research based on the dogfight platform, using enemy AI to train our own agents. However, a series of problems arise when using enemy AI. For example, enemy aircraft will not track us beyond a certain distance. Aircraft, enemy aircraft will fly near the aircraft carrier to attack the aircraft on the aircraft carrier. I sincerely hope to communicate with you.

**Comment by puxd** [comment]
> > > 如果你在每集的开头打开，我认为不会有问题。是的，我打开了 IA 并在任务的每个回合开始前按照你的意愿进行了训练。但经过几轮训练后，它仍然没有跟踪，如下图所示，其中红色代表算法控制的智能，灰色代表IA控制的飞机。
> > 
> > 
> > ![1](https://private-user-images.githubusercontent.com/53037406/309471198-449c8c5e-41bf-407a-b0f1-2d617cb6ead0.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA0MjQxNDAsIm5iZiI6MTcxMDQyMzg0MCwicGF0aCI6Ii81MzAzNzQwNi8zMDk0NzExOTgtNDQ5YzhjNWUtNDFiZi00MDdhLWIwZjEtMmQ2MTdjYjZlYWQwLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE0VDEzNDQwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThlYThmNzk1ZjA0Yzk2ZThkNzQ3MWRkZWMwMzc4Njc3MDk0MjVhZDBjYjUyZmEzMTU1YTEzZTNkNTgyMjIyNWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Z2h6D2FrO6GqaX1AHQ27_7ki6xitg0KLTY6RhiT2wjc)
> > ![2](https://private-user-images.githubusercontent.com/53037406/309471203-5eb64f22-2bf7-4755-a0a5-0baba887d6a1.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA0MjQxNDAsIm5iZiI6MTcxMDQyMzg0MCwicGF0aCI6Ii81MzAzNzQwNi8zMDk0NzEyMDMtNWViNjRmMjItMmJmNy00NzU1LWEwYTUtMGJhYmE4ODdkNmExLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE0VDEzNDQwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI4NzU4YWNjYmQyZDdjZmNmYjM1ZjdhZWNkMjFkNDdlOTZiYjRlNmFmMzMyNTQ1NDA1MGViZGM1OTJhY2Q1NGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-HXglb7XPMA_zrfDiADuu2WEc454Qa-k9J0NDUtkV24)
> 
> 您的问题解决了吗？我还在做基于混战平台的强化学习研究，使用敌方人工智能来训练我们自己的智能体。然而，在使用敌方AI时会出现一系列问题。例如，敌机不会在一定距离之外跟踪我们。飞机，敌机会飞到航母附近，攻击航母上的飞机。我真诚地希望与您交流。



> > > I don't think there will be a problem if you turn on at the beginning of each episode.
> > > Yes, I turned on IA and trained before the start of each turn of the mission like you wanted. But it is still not tracking after some rounds of training, as shown in the image below, where red represents the algorithm-controlled intelligences and gray represents the IA-controlled aircraft.（The second image represents the trajectory of IA when it goes wrong.）
> > 
> > 
> > ![1](https://private-user-images.githubusercontent.com/53037406/309471198-449c8c5e-41bf-407a-b0f1-2d617cb6ead0.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA0MjQxNDAsIm5iZiI6MTcxMDQyMzg0MCwicGF0aCI6Ii81MzAzNzQwNi8zMDk0NzExOTgtNDQ5YzhjNWUtNDFiZi00MDdhLWIwZjEtMmQ2MTdjYjZlYWQwLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE0VDEzNDQwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThlYThmNzk1ZjA0Yzk2ZThkNzQ3MWRkZWMwMzc4Njc3MDk0MjVhZDBjYjUyZmEzMTU1YTEzZTNkNTgyMjIyNWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Z2h6D2FrO6GqaX1AHQ27_7ki6xitg0KLTY6RhiT2wjc)
> > ![2](https://private-user-images.githubusercontent.com/53037406/309471203-5eb64f22-2bf7-4755-a0a5-0baba887d6a1.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA0MjQxNDAsIm5iZiI6MTcxMDQyMzg0MCwicGF0aCI6Ii81MzAzNzQwNi8zMDk0NzEyMDMtNWViNjRmMjItMmJmNy00NzU1LWEwYTUtMGJhYmE4ODdkNmExLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE0VDEzNDQwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI4NzU4YWNjYmQyZDdjZmNmYjM1ZjdhZWNkMjFkNDdlOTZiYjRlNmFmMzMyNTQ1NDA1MGViZGM1OTJhY2Q1NGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-HXglb7XPMA_zrfDiADuu2WEc454Qa-k9J0NDUtkV24)
> 
> Is your problem solved? I am also doing reinforcement learning research based on the dogfight platform, using enemy AI to train our own agents. However, a series of problems arise when using enemy AI. For example, enemy aircraft will not track us beyond a certain distance. Aircraft, enemy aircraft will fly near the aircraft carrier to attack the aircraft on the aircraft carrier. I sincerely hope to communicate with you.

Sure. This is a question I still haven't addressed, and I thought we could communicate about it.
